pkg/segment/sortindex/sortindex.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package sortindex
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/siglens/siglens/pkg/segment/reader/segread/segreader"
+	segutils "github.com/siglens/siglens/pkg/segment/utils"
+	"github.com/siglens/siglens/pkg/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
+)
+
+type Block struct {
+	BlockNum uint16   `json:"blockNum"`
+	Records  []uint16 `json:"records"`
+}
+
+func getFilename(segkey string, cname string) string {
+	return filepath.Join(segkey, cname+".sort")
+}
+
+func WriteSortIndex(segkey string, cname string) error {
```

Comment:
nit: once we finalize the file format, we should write a comment above this func about how the file format looks like